### ISSUE TITLE
[CI] Failure :qa:repository-old-versions:oldES5_0_0Fixture

### ISSUE THREAD

**Issue by andreidan** [elastic/elasticsearch#81980]
**Build scan**:
https://gradle-enterprise.elastic.co/s/2wgivpfwyb2y6

**Repro line**:
```
:qa:repository-old-versions:oldES5_0_0Fixture
```

**Reproduces locally?**: No

**Applicable branches**: 8.0

**Failure excerpt**:
```

11:06:15   [run]
11:06:15     "/var/lib/jenkins/.java/openjdk17/bin/java" "$@" > run.log 2>&1 ; if [ $? != 0 ]; then touch run.failed; fi
11:06:15 oldES5_0_0Fixture output:
11:06:15 -----------------------------------------
11:06:15   failure marker exists: false
11:06:15   pid file exists: false
11:06:15   ports file exists: false
11:06:15 
11:06:15   [log]
11:06:15     Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Dsun.jnu.encoding=UTF8
11:06:15     Running [/private/var/lib/jenkins/workspace/elastic+elasticsearch+8.0+multijob+platform-support-darwin/qa/repository-old-versions/build/tmp/unzipEs5_0_0/elasticsearch-5.0.0/bin/elasticsearch, -p, pid]
11:06:15     Warning: Ignoring JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF8 -Dsun.jnu.encoding=UTF8
11:06:15     Please pass JVM parameters via ES_JAVA_OPTS instead
11:06:15     [2021-12-21T02:05:28,063][INFO ][o.e.n.Node               ] [] initializing ...
11:06:15     [2021-12-21T02:05:29,296][INFO ][o.e.e.NodeEnvironment    ] [z9G7jwK] using [1] data paths, mounts [[/System/Volumes/Data (/dev/disk1s1)]], net usable_space [263.9gb], net total_space [465.4gb], spins? [unknown], types [apfs]
11:06:15     [2021-12-21T02:05:29,297][INFO ][o.e.e.NodeEnvironment    ] [z9G7jwK] heap size [1.9gb], compressed ordinary object pointers [true]
11:06:15     [2021-12-21T02:05:29,300][INFO ][o.e.n.Node               ] [z9G7jwK] node name [z9G7jwK] derived from node ID; set [node.name] to override
11:06:17     [2021-12-21T02:05:29,316][INFO ][o.e.n.Node               ] [z9G7jwK] version[5.0.0], pid[77161], build[253032b/2016-10-26T04:37:51.531Z], OS[Mac OS X/10.15.7/x86_64], JVM[Temurin/OpenJDK 64-Bit Server VM/1.8.0_302/25.302-b08]
11:06:17     Found pid:  77161
11:06:17     [2021-12-21T02:05:37,975][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [aggs-matrix-stats]
11:06:17     [2021-12-21T02:05:37,977][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [ingest-common]
11:06:17     [2021-12-21T02:05:37,977][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [lang-expression]
11:06:17     [2021-12-21T02:05:37,977][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [lang-groovy]
11:06:17     [2021-12-21T02:05:37,978][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [lang-mustache]
11:06:17     [2021-12-21T02:05:37,978][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [lang-painless]
11:06:17     [2021-12-21T02:05:37,978][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [percolator]
11:06:18     [2021-12-21T02:05:37,979][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [reindex]
11:06:18     [2021-12-21T02:05:37,979][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [transport-netty3]
11:06:18     [2021-12-21T02:05:37,980][INFO ][o.e.p.PluginsService     ] [z9G7jwK] loaded module [transport-netty4]
11:06:18     [2021-12-21T02:05:37,981][INFO ][o.e.p.PluginsService     ] [z9G7jwK] no plugins loaded
11:06:18 -----------------------------------------
11:06:35 
11:06:35 > Task :qa:repository-old-versions:oldES5_0_0Fixture FAILED
```

**Comment by mark-vieira** [comment]
Seems to be specific to the mac agents. We're experimenting with different workers here but we might need to bump timeouts in the meantime if we continue to see this.

**Comment by ggsjyoon** [comment]
I'm not sure if an error I experienced, while running `./gradlew check --info` in both `master` and `8.0` branches, is related to the issue but I hope this info helps you in debugging.

```
Resource missing. [HTTP GET: https://repo.maven.apache.org/maven2/adoptium_8/mac/8u302/mac-8u302.pom]
Resource missing. [HTTP GET: https://repo.maven.apache.org/maven2/adoptium_8/mac/8u302/mac-8u302.pom]
Resource missing. [HTTP HEAD: https://api.adoptium.net/v3/binary/version/jdk8u302-b08/mac/aarch64/jdk/hotspot/normal/adoptium]
FAILURE: Build failed with an exception.
* What went wrong:
Could not determine the dependencies of task ':modules:reindex:oldEs2Fixture'.
> Could not resolve all task dependencies for configuration ':modules:reindex:jdk_legacy'.
   > Could not find adoptium_8:mac:8u302.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/adoptium_8/mac/8u302/mac-8u302.pom
       - https://api.adoptium.net/v3/binary/version/jdk8u302-b08/mac/aarch64/jdk/hotspot/normal/adoptium
     Required by:
         project :modules:reindex
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 2m 5s
```

From syncing...
```
Elasticsearch Build Hamster says Hello!
  Gradle Version        : 7.3.2
  OS Info               : Mac OS X 12.1 (aarch64)
  JDK Version           : 17 (Eclipse Adoptium)
  JAVA_HOME             : /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home
  Random Testing Seed   : 198ADA0443E2254B
  In FIPS 140 mode      : false
```

**Comment by mark-vieira** [comment]
@ggsjyoon your issue is that Adoptium doesn't provide Java 8 runtimes for the M1 mac. These tests cannot currently run on aarch64 darwin systems.

**Comment by ap-kulkarni** [comment]
@mark-vieira Azul Zulu provides openjdk 8 distribution for m1 macs. I was able to make necessary changes in build files to fetch that instead of adoptium (for this specific platform only). If there aren't any licensing issues in using the zulu distribution, I would like to open PR for the same. Apologies, since this is not related to the original issue but couldn't find appropriate place to post this.

**Comment by pugnascotia** [comment]
@ap-kulkarni Adoptium and OpenJDK both now provide `aarch64` builds for Mac OS. We switched to the latest OpenJDK release in #86554.

**Comment by ap-kulkarni** [comment]
@pugnascotia This is specific to JDK 8 distribution for MacOS aarch64. Currently only Azul Zulu provides distribution for JDK8 for the said platform.

**Comment by mark-vieira** [comment]
Running these tests on Apple silicon Macs has been fixed in #87733.
